server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMessageListMethod.java

```diff
@@ -123,9 +124,11 @@ private GetMessageListResponse getMessageListResponse(GetMessageListRequest mess
         GetMessageListResponse.Builder builder = GetMessageListResponse.builder();
         try {
             MultimailboxesSearchQuery searchQuery = convertToSearchQuery(messageListRequest);
-            mailboxManager.search(searchQuery,
+            List<MessageId> messageIdList = mailboxManager.search(searchQuery,
                 mailboxSession,
-                messageListRequest.getLimit().orElse(maximumLimit) + messageListRequest.getPosition())
+                messageListRequest.getLimit().orElse(maximumLimit) + messageListRequest.getPosition());
+            builder.total(messageIdList.size());
```

Comment:
You can not do that!

There is a limit, then total is not the size of the returned ids (which is bound by the limit) but the total number of matching results.

Hence the need of search-backend modifications and the relevant implementation advices in the ticket.